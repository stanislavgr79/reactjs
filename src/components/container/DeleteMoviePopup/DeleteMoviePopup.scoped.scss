.site_name {
    padding-top: 10px;
    margin: unset;
}

.delete_movie {
    padding-top: 60px;
    padding-bottom: 40px;
    margin-bottom: 100%;
    .form {
        padding-left: 7%;
        padding-right: 7%;
        width: 60%;
        background-color: #232323;
        position: relative;
        padding-top: 50px;
        .page_title {
            font-size: 36px;
            color: #FFF;
            font-weight: 100;
        }
        .page_description {
            font-size: 22px;
            color: #FFF;
            font-weight: 100;
        }
        .section_icon-close {
            position: absolute;
            right: 20px;
            top: 20px;
        }
        .button_section {
            display: flex;
            flex-direction: row-reverse;
            padding-top: 40px;
            padding-bottom: 60px;
        }
    }
}

.popup_model {
    position: fixed;
    inset: 0px;
    display: flow-root;
    z-index: 999;
    background: rgba(49, 49, 49, 0.88);
    pointer-events: auto;
}

.model_content {
    height: 100%;
    position: relative;
    margin: auto;
    pointer-events: auto;
    .footer {
        position: absolute;
        left: 0;
        bottom: 0;
        width: 100%;
    }
}