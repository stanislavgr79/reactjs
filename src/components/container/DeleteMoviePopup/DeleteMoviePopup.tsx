import React, { useCallback } from 'react';
import { useDispatch } from 'react-redux';
import { fetchDeleteMovie, updateShowPopup } from '../../../redux/actions/movies-actions';

import Sitename from '../../atom/SiteName';
import Button from '../../atom/Button';
import Footer from '../../structure/Footer';

import './DeleteMoviePopup.scoped.scss';

interface IProps {
  movieId?: number;
  closePopup: () => void;
}

export default function DeleteMoviePopup({ movieId, closePopup }: IProps): JSX.Element {
  const dispatch = useDispatch();

  const handleSubmit = useCallback(() => {
    dispatch(fetchDeleteMovie(movieId));
    dispatch(updateShowPopup(false));
    closePopup();
  }, [dispatch, movieId, closePopup]);

  const handleClick = useCallback(() => {
    dispatch(updateShowPopup(false));
    closePopup();
  }, [dispatch, closePopup]);

  return (
    <div className="popup_model">
      <div className="model_content container-md">
        <Sitename />
        <div className="container delete_movie">
          <div className="container form">
            <div className="section_icon-close">
              <Button
                buttonType="button"
                id="delete_movie_btn_close"
                className="btn_close"
                onClick={handleClick}
              />
            </div>
            <div className="page_title">DELETE MOVIE</div>
            <div className="page_description">Are you sure want to delete this movie?</div>
            <div className="button_section">
              <Button
                buttonType="submit"
                className="btn_submit"
                id="delete_movie_btn_submit"
                label="CONFIRM"
                onClick={handleSubmit}
              />
            </div>
          </div>
        </div>
        <Footer />
      </div>
    </div>
  );
}
