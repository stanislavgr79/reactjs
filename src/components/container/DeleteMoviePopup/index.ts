import DeleteMoviePopup from './DeleteMoviePopup';

export default DeleteMoviePopup;
