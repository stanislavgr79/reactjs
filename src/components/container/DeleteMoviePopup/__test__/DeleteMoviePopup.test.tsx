import React from 'react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';

import DeleteMoviePopup from '../DeleteMoviePopup';

import { render, screen, fireEvent } from '@testing-library/react';
import {
  DELETE_MOVIE_SUCCESS,
  MoviesActionTypes,
  UPDATE_STATUS_MOVIE_POPUP,
} from '../../../../redux/types/movies-reducer-types';

jest.mock('../../../atom/SiteName', () => {
  return {
    __esModule: true,
    default: () => {
      return <div className="siteName"></div>;
    },
  };
});
jest.mock('../../../structure/Footer', () => {
  return {
    __esModule: true,
    default: () => {
      return <div className="footer"></div>;
    },
  };
});
jest.mock('../../../../redux/actions/movies-actions', () => ({
  fetchDeleteMovie: (): MoviesActionTypes => {
    return {
      type: DELETE_MOVIE_SUCCESS,
    };
  },
  updateShowPopup: (): MoviesActionTypes => {
    return {
      type: UPDATE_STATUS_MOVIE_POPUP,
      payload: false,
    };
  },
}));

describe('DeleteMoviePopup', () => {
  const initialState = { moviesStore: { showPopup: true } };
  const mockStore = configureStore();
  const movieId = 123456;
  const closePopup = jest.fn();
  let store;

  afterEach(() => {
    jest.clearAllMocks();
  });

  test('handleClick function', () => {
    store = mockStore(initialState);
    const tree = render(
      <Provider store={store}>
        <DeleteMoviePopup movieId={movieId} closePopup={closePopup} />
      </Provider>,
    );

    const expectedActions = [{ type: 'UPDATE_STATUS_MOVIE_POPUP', payload: false }];

    expect(tree.baseElement).toMatchSnapshot();
    fireEvent.click(screen.getByRole('button', { name: '' }));
    expect(closePopup).toBeCalled();
    expect(store.getActions()).toEqual(expectedActions);
  });

  test('handleSubmit function', () => {
    store = mockStore(initialState);
    const tree = render(
      <Provider store={store}>
        <DeleteMoviePopup movieId={movieId} closePopup={closePopup} />
      </Provider>,
    );

    const expectedActions = [
      { type: 'DELETE_MOVIE_SUCCESS' },
      { type: 'UPDATE_STATUS_MOVIE_POPUP', payload: false },
    ];

    expect(tree.baseElement).toMatchSnapshot();
    fireEvent.click(screen.getByRole('button', { name: 'CONFIRM' }));
    expect(closePopup).toBeCalled();
    expect(store.getActions()).toEqual(expectedActions);
  });
});
