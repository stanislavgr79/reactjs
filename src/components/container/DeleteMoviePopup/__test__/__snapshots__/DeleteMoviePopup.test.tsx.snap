// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteMoviePopup handleClick function 1`] = `
<body>
  <div>
    <div
      class="popup_model"
    >
      <div
        class="model_content container-md"
      >
        <div
          class="siteName"
        />
        <div
          class="container delete_movie"
        >
          <div
            class="container form"
          >
            <div
              class="section_icon-close"
            >
              <button
                class="btn button btn_close"
                id="delete_movie_btn_close"
                type="button"
              />
            </div>
            <div
              class="page_title"
            >
              DELETE MOVIE
            </div>
            <div
              class="page_description"
            >
              Are you sure want to delete this movie?
            </div>
            <div
              class="button_section"
            >
              <button
                class="btn button btn_submit"
                id="delete_movie_btn_submit"
                type="submit"
              >
                CONFIRM
              </button>
            </div>
          </div>
        </div>
        <div
          class="footer"
        />
      </div>
    </div>
  </div>
</body>
`;

exports[`DeleteMoviePopup handleSubmit function 1`] = `
<body>
  <div>
    <div
      class="popup_model"
    >
      <div
        class="model_content container-md"
      >
        <div
          class="siteName"
        />
        <div
          class="container delete_movie"
        >
          <div
            class="container form"
          >
            <div
              class="section_icon-close"
            >
              <button
                class="btn button btn_close"
                id="delete_movie_btn_close"
                type="button"
              />
            </div>
            <div
              class="page_title"
            >
              DELETE MOVIE
            </div>
            <div
              class="page_description"
            >
              Are you sure want to delete this movie?
            </div>
            <div
              class="button_section"
            >
              <button
                class="btn button btn_submit"
                id="delete_movie_btn_submit"
                type="submit"
              >
                CONFIRM
              </button>
            </div>
          </div>
        </div>
        <div
          class="footer"
        />
      </div>
    </div>
  </div>
</body>
`;
